dashboard/app/models/user.rb

```diff
@@ -1981,6 +1981,13 @@ def undestroy
     restore(recursive: true, recovery_window: 5.minutes)
   end
 
+  def depended_upon_for_login?
+    return false unless teacher?
+    students.any? do |student|
```

Comment:
Not saying `return false unless teacher?` is a bad idea, but I think it's technically redundant in this case; `students` should be empty for anyone that's not a teacher, because `Follower` validates:

https://github.com/code-dot-org/code-dot-org/blob/2d0981ed044603314ab9463eea94d8288924c23b/dashboard/app/models/follower.rb#L44